src/main/java/bio/overture/ego/service/LinkedinService.java

```diff
@@ -0,0 +1,76 @@
+package bio.overture.ego.service;
+
+import static bio.overture.ego.model.exceptions.InternalServerException.buildInternalServerException;
+import static java.util.Collections.singletonMap;
+import static org.springframework.http.HttpStatus.OK;
+
+import bio.overture.ego.model.dto.LinkedinEmail;
+import bio.overture.ego.model.dto.LinkedinEmailResponse;
+import bio.overture.ego.model.enums.LinkedinContactType;
+import java.util.Map;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.oauth2.client.OAuth2RestOperations;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpStatusCodeException;
+
+@Slf4j
+@Service
+public class LinkedinService {
+
+  private String userEmailUri;
+
+  @Autowired
+  public LinkedinService(@NonNull @Value("${linkedIn.resource.userEmailUri}") String userEmailUri) {
+    this.userEmailUri = userEmailUri;
+  }
+
+  public Map<String, Object> getPrimaryEmail(
+      OAuth2RestOperations restTemplate, @NonNull Map<String, Object> map) {
+    val headers = new HttpHeaders();
+    headers.set("Accept", "application/json");
```

Comment:
Spring constant for `"Accept"`: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/HttpHeaders.html#ACCEPT

Reply:
Spring constant for `"application/json"`: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/MediaType.html#APPLICATION_JSON